packages/react/src/reactrouterv3.ts

```diff
@@ -150,9 +150,17 @@ function getRouteStringFromRoutes(routes: Route[]): string {
     }
   }
 
-  return routesWithPaths
+  const pathParts = routesWithPaths
     .slice(index)
     .filter(({ path }) => !!path)
-    .map(({ path }) => path)
-    .join('');
+    .map(({ path }) => path);
+
+  // Join all parts with '/', then replace multiple slashes with a single one.
+  let fullPath = pathParts.join('/');
+  fullPath = fullPath.replace(/\/+/g, '/');
```

Comment:
We should check if `reduce` is faster than the join + replace here.

it will also likely be more bundle size efficient, and we can account for the edge case.

Reply:
thought about this more - I think this works. We can replace both `const pathParts` and the `fullPath` calculations.

```
return routesWithPaths.slice(index).reduce((acc, { path }) => {
  const pathSegment = acc === '/' || acc === '' ? path : `/${path}`;
  return `${acc}${pathSegment}`;
}, '');
```

Reply:
Benchmark: https://jsbench.me/g2mamtp9bd/1

Approach reduce is 5% faster but it makes the code harder to read and maintain and I don't think it helps with the bundle size.

Let me know what you think.

Reply:
Reduce is 2.51% slower: https://jsbench.me/g2mamtp9bd/1

Reply:
On my M1 Macbook it does 102K ops/s, I don't think it matters that much. Added a commit here: https://github.com/getsentry/sentry-javascript/pull/16274/commits/c1144e2734510ca6e38406499781b93c5718ea71